pkg/api/validation/validation_test.go

```diff
@@ -3702,54 +3819,6 @@ func TestValidatePod(t *testing.T) {
 			},
 			Spec: validPodSpec(nil),
 		},
-		{ // valid opaque integer resources for init container
```

Comment:
`ValidatePod` will invoke `ValidateContainers`, so we don't need these test cases here. Move them to `TestValidateContainers`. And we don't need to validate init container and regular container separately there.